src/main/java/com/spring/assignment/employeesystem/annotation/UserUniqueValidator.java

```diff
@@ -0,0 +1,42 @@
+package com.spring.assignment.employeesystem.annotation;
+
+import com.spring.assignment.employeesystem.dao.EmployeeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+@Service
+public class UserUniqueValidator implements ConstraintValidator<Unique,String> {
+
+   private EmployeeRepository employeeRepository;
+    public UserUniqueValidator(){}
+
+    @Autowired
+    public UserUniqueValidator(EmployeeRepository employeeRepository){
+        this.employeeRepository=employeeRepository;
+    }
+
+
+    @Override
+    public void initialize(Unique constraintAnnotation) {
+        constraintAnnotation.message();
+    }
+
+    @Override
+    public boolean isValid(String username, ConstraintValidatorContext constraintValidatorContext) {
```

Comment:
You can refactor this method by declaring flag outside the if loop and default value as true and just change the value of flag inside if loof and directly flag